### PR TITLE
[Misc] Support FP8 kv cache scales from compressed-tensors

### PR DESCRIPTION
**Pull request by mgoin**
Adding the logic for loading models with quantized kv cache scales, generated using `compressed-tensors` framework.

- `CompressedTensorsConfig` now has an optional `kv_cache_scheme` argument. As of the next `compressed-tensors` release, the key contains the information about the properties of the quantized kv cache. 
- Added an interface `BaseKVCacheMethod` that both `Fp8KVCacheMethod` and `CompressedTensorsKVCacheMethod` inherits from, to prepare the `k_scale` and `v_scale` attributes of the quantized `Attention` layer.

Currently, the loading of the kv cache scales happens inside the `load_model` method of the model (in this PR only LLaMa model), just like previous kv scales. However, it needs to happen before the parameter names get mapped to stacked params since we are looking for `k_proj.output_scale` and `v_proj.output_scale`. Thich is a bit ugly because we need to copy and paste the helper function into every model to make it read quantized kv cache.

Done in collaboration with @dbogunowicz

### CODE REVIEW

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py [hunk 1]
```diff
@@ -309,3 +318,49 @@ def apply(self,
...
+                f"Expected strategy: tensor, found strategy: {strategy}")
+
+        is_symmetric = kv_cache_scheme.get("symmetric")
+        if not is_symmetric:
+            raise NotImplementedError(
+                "Only support symmetric scaling factor "
+                "for compressed-tensors KV cache. "
+                f"However found symmetric: {is_symmetric}")
+
+        return
```
Comment (comaniac) [inline vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py:366]: Not necessary return.

File: vllm/model_executor/layers/quantization/compressed_tensors/utils.py [hunk 2]
```diff
@@ -209,6 +209,23 @@ def _find_first_match(value: str,
...
+    Check whether the param name matches the format for k/v cache scales
+    in compressed-tensors. If this is the case, return its equivalent
+    param name expected by vLLM
+
+    :param name: param name
+    :return: matching param name for KV cache scale in vLLM
+    """
+    if name.endswith(".output_scale") and ".k_proj" in name:
+        return name.replace(".k_proj.output_scale", ".attn.k_scale")
+    elif name.endswith(".output_scale") and ".v_proj" in name:
```
Comment (comaniac) [inline vllm/model_executor/layers/quantization/compressed_tensors/utils.py:223]: nit: Use `if` when each if-branch returns.
```suggestion
    if name.endswith(".output_scale") and ".v_proj" in name:
```

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py [hunk 3]
```diff
@@ -309,3 +318,49 @@ def apply(self,
         if scheme is None:
             raise ValueError("A scheme must be defined for each layer")
         return scheme.apply_weights(layer, x, bias=bias)
+
+
+class CompressedTensorsKVCacheMethod(BaseKVCacheMethod):
+    """
+    Supports loading kv-cache scaling factors from compressed-tensors
+    checkpoints.
+    """
```
Comment (comaniac) [inline vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py:327]: Is this class completed? The only difference between this and Fp8KVCacheMethod seems just an additional validator?

**Review by comaniac — APPROVED** [review]
Overall LGTM

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py (hunk 3 above)
Comment (mgoin) [reply vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py:327]: Yeah it's just a start of structure for future work. We could support integer or other scaling schemes within each method, so wanted a base class to bind them to.

**Pull request merged**